diff --git a/arch/i386/Makefile b/arch/i386/Makefile
index 425361f..9e8271c 100644
--- a/arch/i386/Makefile
+++ b/arch/i386/Makefile
@@ -4,7 +4,7 @@
 
 # Allow i486 insns (basically, bswap insn)
 # Do not try to tune for 486+ (might add padding)
-CFLAGS += $(call cc-option,-march=i486 -mtune=i386,)
+CFLAGS += $(call cc-option,-mpreferred-stack-boundary=2,)
 
 ifeq ($(CONFIG_STACK_OPTIMIZATION_386),y)
 # -mpreferred-stack-boundary=2 is essential in preventing gcc 4.2.x
